NeoML/src/Dnn/Layers/ParameterLayer.cpp

```diff
@@ -0,0 +1,81 @@
+/* Copyright © 2017-2023 ABBYY Production LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#include <common.h>
+#pragma hdrstop
+
+#include <NeoML/Dnn/Layers/ParameterLayer.h>
+#include <NeoMathEngine/NeoMathEngine.h>
+
+namespace NeoML {
+
+void CParameterLayer::AllocateOutputBlobs()
+{
+	outputBlobs[0] = paramBlobs[0];
+}
+
+void CParameterLayer::SetBlob(CDnnBlob* _blob)
+{
+	if (_blob == paramBlobs[0].Ptr()) {
```

Comment:
Maybe, I'm not sure, the check should also be here, that blob.ReinterpretDimensions() on the changing blob is not be called outside. And the blob still has the same dimensions.